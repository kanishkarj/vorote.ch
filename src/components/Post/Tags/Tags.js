// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import styled from 'styled-components';
import { marginBottom } from '../../../utils/get-margin';

const StyledTags = styled.div`
  ${marginBottom(0.5)}
`;

const TagsList = styled.ul`
  list-style: none;
  margin: 0 -10px;
  padding: 0;
`;

const TagsListItem = styled.li`
  display: inline-block;
  margin: 10px 5px;
`;

const TagLink = styled(Link)`
  display: inline-block;
  height: ${(props) => props.theme.buttons.height}px;
  padding: 0 24px;
  line-height: ${(props) => props.theme.buttons.height}px;
  border: 1px solid var(--color-gray500);
  text-decoration: none;
  border-radius: ${(props) => props.theme.buttons.borderRadius}px;
  color: var(--color-text);

  &:hover,
  &:focus {
    color: var(--color-primary);
  }
`;

type Props = {
  tags: string[],
  tagSlugs: string[]
};

const Tags = ({ tags, tagSlugs }: Props) => (
  <StyledTags>
    <TagsList>
      {tagSlugs && tagSlugs.map((slug, i) => (
        <TagsListItem key={tags[i]}>
          <TagLink to={slug}>{tags[i]}</TagLink>
        </TagsListItem>
      ))}
    </TagsList>
  </StyledTags>
);

export default Tags;
