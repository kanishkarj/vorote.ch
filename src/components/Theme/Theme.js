import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  fonts: {
    body: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    heading: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    monospace: 'Menlo, monospace',
  },
  fontWeights: {
    body: 400,
    heading: 600,
    bold: 700,
  },
  fontSizes: {
    base: 16,
    small: 14,
  },
  lineHeight: 1.625,
  spacing: Math.round(16 * 1.625),
  buttons: {
    height: 35,
    borderRadius: 20,
  },
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  }
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

export default Theme;
