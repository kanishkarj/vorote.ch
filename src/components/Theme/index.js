// @flow strict
export { default } from './Theme';
