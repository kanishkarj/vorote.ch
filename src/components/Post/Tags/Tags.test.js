// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Tags from './Tags';
import 'jest-styled-components';
import Theme from '../../Theme';


describe('Tags', () => {
  it('renders correctly', () => {
    const props = {
      tags: [
        'test_0',
        'test_1'
      ],
      tagSlugs: [
        '/test_0',
        '/test_1'
      ]
    };

    const tree = renderer.create(
    <Theme>
      <Tags {...props} />
    </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
