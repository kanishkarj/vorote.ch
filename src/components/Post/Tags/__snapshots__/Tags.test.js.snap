// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
.c0 {
  margin-bottom: 13px;
}

.c1 {
  list-style: none;
  margin: 0 -10px;
  padding: 0;
}

.c2 {
  display: inline-block;
  margin: 10px 5px;
}

.c3 {
  display: inline-block;
  height: 35px;
  padding: 0 24px;
  line-height: 35px;
  border: 1px solid var(--color-gray500);
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 20px;
  color: var(--color-text);
}

.c3:hover,
.c3:focus {
  color: var(--color-primary);
}

<div
  className="c0"
>
  <ul
    className="c1"
  >
    <li
      className="c2"
    >
      <a
        className="c3"
        href="/test_0"
      >
        test_0
      </a>
    </li>
    <li
      className="c2"
    >
      <a
        className="c3"
        href="/test_1"
      >
        test_1
      </a>
    </li>
  </ul>
</div>
`;
